server/tests/plugin/auror_hostnames/test_core.py

```diff
@@ -0,0 +1,78 @@
+# This file is part of sner4 project governed by MIT license, see the LICENSE.txt file.
+"""
+auror_hostnames plugin core tests
+"""
+
+from unittest.mock import mock_open
+import os
+import subprocess
+
+import dns.zone
+
+from sner.plugin.auror_hostnames.core import clone_dns_repo, get_zone_file_paths, process_cnames, process_ptrs, get_records, get_repos
+
+
+def test_get_repos(mocker):
+    """Test getting DNS repositories"""
+    mocker.patch("os.getenv", return_value="dummy_token")
+    mocker.patch("subprocess.check_output", return_value="dummy_repo1\t\ndummy_repo2\t\n")
+    result = get_repos()
+    assert result == ["dummy_repo1", "dummy_repo2"]
+
+
+def test_clone_dns_repo(mocker):
+    """Test cloning DNS repo"""
+    mocker.patch("os.environ.copy", return_value={"GIT_SSH_COMMAND": "dummy_command"})
+    mocker.patch("subprocess.run")
+    clone_dns_repo("dummy_git_server", "dummy_repo", "dummy_key")
+    os.environ.copy.assert_called_once()
```

Comment:
can we assign mocker/spy into variable and make assertions on that var? only for readability